### PR TITLE
fix(locale): improve en_GB location addresses

### PR DESCRIPTION
**Pull request by matthewmayer**
Improved addresses for en_GB locale. The default en locale uses quite Americanized street and city prefixes/suffixes

## Examples in en_US

```
'034 Joshua Parkway, Port Theatown',
  '552 Oliver Pine, New Kathleenland',
  '55436 Jerod Canyon, Connmouth',
  '315 Kellen Extensions, West Anibalhaven',
  '711 Brayan Trafficway, Josephinechester',
  '8512 Padberg Stravenue, Hauckstad',
  '2970 Lilyan Mall, Selinachester',
  '1752 Donnelly Underpass, Lake Marianna',
  '29901 Cicero Cove, Fort Oliver',
  '5195 Maye Dale, Grimesfort',
  '758 Aaron Lodge, West Mario',
  '0544 Lilliana Drives, West June',
  '303 Reymundo Walks, North Rosamond',
  '6244 Heller Spring, West Des Moines',
  '2680 Little Crossing, Rolfsonshire',
  '64131 Brooke Mission, Port Aubrey',
  '5835 Bernhard Overpass, Lake Cameron',
  '954 Euna Light, Santee',
  '032 Rath Freeway, Fort Isidro',
  '96845 Swift Passage, Fort Arely',
  '092 Marjolaine Villages, Gottliebberg',
  '098 Jakubowski Rue, MacGyverboro',
  '9080 Fadel Oval, South Lorenz',
  '3026 Glover Views, Port Jerry',
  '36144 Braun Walks, Thoraton',
  '711 Eleonore Place, Aufderharboro',
  '414 Kris Squares, Fort Nia',
  '1056 Leda Camp, Lancaster',
  '2483 Tillman Ford, Palm Coast',
  '18226 Gretchen Harbors, West Olen',
  '20481 Sylvester Avenue, Port Paula',
  '00581 Earline Motorway, Bergnaumland',
  '7744 Schmitt Light, Quitzonhaven',
  '1163 Piper Green, East Easterland',
  '214 Murazik Wall, Fort Simone',
  '009 Murphy Freeway, South Scarlett',
  '0696 Amanda Walks, West Rogersboro',
  '72723 Vandervort Wall, Hermistonfield',
  '8647 Loma Expressway, North Connie',
  '052 Bernier Hill, New Elyssa',
  '89175 Myrtice Village, North Tylerworth',
  '7628 Kuvalis Extensions, West Lorachester',
  '397 Cathrine Well, Port Melvinboro',
  '1163 Wunsch Stream, Angelitahaven',
  '48917 Jerrell Manors, South Ismaelfort',
  '1051 Adams Freeway, East Donnystead',
  '551 Evie Port, Ullrichview',
  '04771 Beau Skyway, Clearwater',
  '0880 Wintheiser Meadow, Fort Ally',
  '345 Hyatt Dale, Mesa',
  '453 Amiya Land, Appleton',
  '99786 Ricardo Estate, Berwyn',
  '1064 Hackett Park, Lake Josefa',
  '92041 Nathaniel Overpass, New Britain',
  '76516 Hand Plaza, Mauricestad',
  '97181 Maggio Mills, West Misael',
  '268 Lind Corners, Bakersfield',
  '86253 Morar Bridge, Bertramstead',
  '7513 Braun Loaf, Mesquite',
  '128 Funk Shoals, Fort Theresefort',
  '301 Ortiz Views, Pollytown',
  '04444 Becker Brook, Taylorsville',
  '69717 Casimir Cliff, Bernierborough',
  '35344 Palma Inlet, Edina',
  '08804 Emard Meadows, Port Arvillafield',
  '9741 Kovacek Throughway, East Perry',
  '54480 Ludwig Curve, Fort Molly',
  '680 Becker Forest, New Christelle',
  '314 Dare Pike, Lake Erich',
  '2403 McGlynn Manors, Schowalterworth',
  '5252 Kreiger Corners, Lake Maryamberg',
  '983 Ana Groves, Monroe',
  '54777 Runte Village, Hermanfurt',
  '9079 Else Bridge, Erlington',
  '088 Mekhi Loop, New Coralie',
  '4124 Judah Dale, Hannastad',
  '408 Turner Viaduct, Presleyport',
  '10827 Mae Locks, Faustoland',
  '4901 Jacobi Alley, South Myron',
  '3427 Feeney Ford, New Rochelle',
  '7818 Gino Branch, Apex',
  '5659 Frances Alley, Margate',
  '0227 Wendy Dale, West Davinside',
  '93661 Dickens Oval, Antoniafurt',
  '216 Jillian Creek, Deangeloboro',
  '614 Destini Landing, West Berthaland',
  '520 Melyna Manor, New Fernando',
  '6558 Theo Plain, Cathedral City',
  '496 Barton Drive, Bernadineworth',
  '51959 Keeling Run, Beertown',
  '315 Nash Flats, Nolanfield',
  '37444 Felicita Mountains, South Normaberg',
  '6354 Concepcion Corners, Ludwigstad',
  '8012 Monty Land, Champlinfield',
  '73242 Wiza Highway, Donaldview',
  '16450 Kenna Mount, Lake Cloyd',
  '103 Pansy Extensions, Fort Deltaville',
  '760 Collier Course, Arttown',
  '174 Quitzon Drive, West Jacynthe',
  '14226 Camilla Forge, Schummshire'
```

## Examples in en_GB

```
 '556 Tromp Drive, Labadie-under-Glover',
  '50 Ana Grove, Nether Ernser',
  '909 Alfredo Orchard, Skileshampton',
  '81 Ankunding Hill, Ebert-le-Kutch',
  '45 Lonnie Park, Upton Lemkehampton',
  '075 Cremin Croft, Nolanford',
  '8 Gutkowski Parade, Zboncak-under-Kuhic',
  '89 Dallin Croft, McDermott-upon-Murray',
  '903 Langosh Chase, South Abshire',
  '6 Bailey Way, Bradtkeborough',
  '17 Gregoria Chase, Old Hellerridge',
  '88 Maye End, Lednerwick',
  '865 Stracke Avenue, South Mills Court',
  '6 Treutel Fold, West Luettgen Hill',
  '7 Schroeder Rise, Collins-on-Harvey',
  '061 Flavio Corner, Bartellingham',
  '78 Schroeder Green, Long Daching',
  '802 Monte Mews, Heathcote-under-Bashirian',
  '6 Beer Gate, Nether Altenwerth',
  '870 Anderson Corner, Jones-upon-White',
  '5 Orlando Close, Fahey-upon-Kshlerin',
  '95 Boyle Approach, Castle Bernier',
  '2 Dayton Brow, Great Herzog Park',
  '683 Upton Lane, Bashirian-under-Howe',
  '7 Kutch Fields, Ryanhampton',
  '8 Rahul Way, Douglas-over-Bashirian',
  '1 Verla Mount, Rohan-over-Ortiz',
  '71 Alessia Hill, St. Thielstone',
  '96 Harold Place, Balistreri Place',
  '64 Sabrina Hill, Dach-under-Mayert',
  '72 Cathryn Brow, Brown-on-Goldner',
  '343 Braun Fields, Wyman-on-Johnson',
  '252 Tyler Mead, Mitchell-over-Toy',
  '7 Quigley Ridge, Crona-le-Bradtke',
  '386 Heloise Square, Farrell-le-Rolfson',
  '4 Shawn Wood, Hammes-le-Kemmer',
  '94 Onie Brow, Howe End',
  '478 Abbey End, Great Hegmannbury',
  '08 Ruben Drove, Prosacco-under-Hettinger',
  '9 Anderson Fold, North Nitzsche',
  '06 Damien Lane, Aufderhar-le-Jacobson',
  '625 Becker End, Handdon',
  '12 Maude Grove, Heidenreichley',
  '97 Ziemann Ridge, Stroman-under-Wilkinson',
  '2 Stephan Approach, Hahning',
  '362 Alysha Street, Powlowski-on-Cole',
  '6 Merl Lea, Cassin Bridge',
  "63 O'Conner Parade, Linding",
  '56 Barton Corner, Upper Willmswood',
  '97 Ebba Place, Leffler-on-Hettinger',
  '016 Rhianna Garth, Vonstone',
  '03 Dean Gate, St. Klinghampton',
  '52 Lueilwitz Brow, Old Sanford',
  '2 Clement Croft, Blick-le-Huels',
  '04 Will Vale, Strosinfield',
  '8 Hartmann Brae, Old Upton Gardens',
  '83 Carroll Hill, Schimmel-over-Nader',
  "2 O'Connell Court, Barrows-over-Schulist",
  '318 Smith Hill, North Waters',
  '3 Nakia Copse, Wehnerton',
  '559 Selina Drove, Ziemanndon',
  '572 Wiza Paddock, North Friesenthorpe',
  '264 Modesta Square, Cormier Gardens',
  '566 Kirstin Fields, Upper Spencer Common',
  '2 Wuckert Orchard, St. Welchington',
  '6 Meaghan Green, Hackettington',
  '5 Gulgowski Way, Little Botsford',
  '0 Hammes Orchard, St. Schaden',
  '0 Pedro Park, Marvin-upon-Okuneva',
  '2 Khalid Terrace, Berge Cross',
  '95 Garnet Place, Beahan-on-Dooley',
  '430 Mckayla Wynd, Little Schmitt',
  '0 Dianna Ridge, Prohaska-under-MacGyver',
  '140 Maymie Close, Fahey-on-Robel',
  '22 Wuckert Drove, Castle Bergnaumham',
  '425 Guido Mews, St. Lockman Park',
  '99 Johnston Field, Hirthe Green',
  '4 Brakus Yard, Nether Weissnat',
  '514 Predovic Square, Feest Cross',
  '15 Wehner End, Kreiger-over-Heathcote',
  '479 Alec Approach, Newton McCullough End',
  '3 Otilia Terrace, Newton Wisozkton',
  '4 Rippin Paddock, Goldner-upon-Mayer',
  '68 Marley Wynd, Heller Cross',
  '035 Konopelski Green, Borer-under-Macejkovic',
  '77 Benedict Square, Upton Stoltenbergdon',
  '83 Prohaska Rise, Kub-on-Mitchell',
  '42 Stephon Walk, Macejkovic Bridge',
  '29 Schinner Garth, Upton Kihnham',
  '645 Benjamin Path, Rutherford-under-Turcotte',
  '30 Lowell Bank, Kuhlman-on-Douglas',
  '25 Nikolaus Gate, North Reynolds End',
  '0 Buddy Meadows, Stromanbury',
  '723 Nolan Fields, Lakin Gardens',
  '703 Brenda Street, Nether Cummeratawick',
  '8 Davion Drove, Kerluke-on-Beatty',
  '0 Maybelle View, Schimmel Gardens',
  '3 Fadel Meadows, Gleichnerworth',
  '4 Preston Grange, Wisoky Green',
  '8 Sylvester Rise, Mullerley'
]

```

### CODE REVIEW

**Review by ST-DDT — APPROVED** [review]

File: src/locales/en_GB/location/index.ts [hunk 1]
```diff
@@ -3,19 +3,29 @@
...
 import default_country from './default_country';
 import postcode from './postcode';
 import street from './street';
+import street_suffix from './street_suffix';
 import uk_country from './uk_country';
 
 const location: LocationDefinitions = {
+  building_number,
   city,
+  city_infix,
```
Comment (Shinigami92) [inline src/locales/en_GB/location/index.ts:21]: `city_infix` is new to me, do we have it in any other set?

File: src/locales/en_GB/location/city_suffix.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ' Park',
+  ' Hill',
+  ' Court',
+  ' Heath',
+  ' Bridge',
+  ' End',
+  ' Common',
+  ' Place',
+  ' Cross',
+  ' Gardens',
```
Comment (Shinigami92) [inline src/locales/en_GB/location/city_suffix.ts:32]: mh :thinking: 
looks a bit weird to me that there are values with leading white space
I hope they get not trimmed somewhere

**Review by Shinigami92 — APPROVED** [review]
@matthewmayer You wrote

> The default en locale uses quite Americanized street and city prefixes/suffixes

But there is no explanation why the new patterns are improvements (I'm a non-brithish person and don't have any kind of knowledge about streetnames in GB)

---

Beside my few comments, nothing blocking this PR from getting merged
_just curious_

File: src/locales/en_GB/location/index.ts (hunk 1 above)
Comment (ST-DDT) [reply src/locales/en_GB/location/index.ts:21]: All prefix, infix, suffix variants are locale specific.
(There isn't/shouldn't be a way to access these other than fake patterns, there shouldn't be localized definition tree entries for them either.)

File: src/locales/en_GB/location/city_suffix.ts (hunk 2 above)
Comment (ST-DDT) [reply src/locales/en_GB/location/city_suffix.ts:32]: I'm not sure on how to handle these either.
Maybe using a new city pattern and a separate file name?

**Comment by matthewmayer** [comment]
There's just different historical context for the names

Street numbers tend to be lower
Streets end in particular endings https://towardsdatascience.com/what-can-analysing-more-than-2-million-street-names-reveal-c94be585759
City names follow certain historical patterns https://en.wikipedia.org/wiki/List_of_generic_forms_in_place_names_in_Ireland_and_the_United_Kingdom?wprov=sfti1

And new street names are often mandated to follow old patterns by local governments

https://www.west-norfolk.gov.uk/info/20083/address_management/498/street_naming_conventions

**Pull request merged**